walkbox.go

```diff
@@ -147,6 +147,7 @@ func (wb *WalkBox) distance(p Positionf) float32 {
 // WalkBoxMatrix represents a collection of WalkBoxes and their adjacency relationships.
 type WalkBoxMatrix struct {
 	walkBoxes       []*WalkBox
+	walkBoxesMap    map[string]*WalkBox
```

Comment:
Please don't do this. Having two different structures is a cognitive load that doesn't worth. 

Instead, define a `findWalkbox(id string) *Walkbox` private function that iterates over the `walkBoxes` field and returns the matching walkbox. And use where needed.